sunbeam-python/sunbeam/provider/local/steps.py

```diff
@@ -255,3 +254,286 @@ def _update_microcluster_status(self, status: dict, microcluster_status: dict):
+                "This allows specifying a list of SR-IOV devices that "
+                "will be exposed to Openstack instances."
+            ),
+        ),
+    }
+
+
+class LocalConfigSRIOVStep(BaseStep):
+    """Prompt user for SR-IOV configuration."""
+
+    def __init__(
+        self,
+        client: Client,
+        node_name: str,
+        jhelper: JujuHelper,
+        model: str,
+        manifest: Manifest | None = None,
+        accept_defaults: bool = False,
+        show_initial_prompt: bool = True,
+    ):
+        super().__init__("SR-IOV Settings", "Configure SR-IOV")
+        self.client = client
+        self.node_name = node_name
+        self.jhelper = jhelper
+        self.model = model
+        self.manifest = manifest
+        self.accept_defaults = accept_defaults
+        self.variables: dict = {}
+        # Avoid the "Configure SR-IOV?" question if the user
+        # specifically asked for this.
+        self.show_initial_prompt = show_initial_prompt
+
+    def prompt(
+        self,
+        console: Console | None = None,
+        show_hint: bool = False,
+    ) -> None:
+        """Determines if the step can take input from the user.
+
+        Prompts are used by Steps to gather the necessary input prior to
+        running the step. Steps should not expect that the prompt will be
+        available and should provide a reasonable default where possible.
+        """
+        if not console:
+            LOG.info("No console available, skipping prompt.")
+            return
+
+        self.variables = sunbeam.core.questions.load_answers(
+            self.client, PCI_CONFIG_SECTION
+        )
+
+        pci_whitelist: list[dict] = []
+        excluded_devices: dict[str, list] = {}
+
+        if self.manifest:
+            if (
+                self.manifest.core.config.pci
+                and self.manifest.core.config.pci.device_specs
+            ):
+                pci_whitelist = self.manifest.core.config.pci.device_specs
+                LOG.debug("PCI whitelist from manifest: %s", pci_whitelist)
+            if (
+                self.manifest.core.config.pci
+                and self.manifest.core.config.pci.excluded_devices
+            ):
+                excluded_devices = self.manifest.core.config.pci.excluded_devices
+                LOG.debug("PCI exclude list from manifest: %s", excluded_devices)
+
+        previous_pci_whitelist = self.variables.get("pci_whitelist") or []
+        previous_excluded_devices = self.variables.get("excluded_devices") or {}
+
+        LOG.debug("PCI whitelist from previous answers: %s", previous_pci_whitelist)
+        LOG.debug(
+            "PCI exclude list from previous answers: %s", previous_excluded_devices
+        )
+
+        if self.node_name not in excluded_devices:
+            excluded_devices[self.node_name] = []
+
+        if self.accept_defaults:
+            LOG.debug(
+                "Received '--accept-defaults', ignoring previous "
```

Comment:
Need to check, but accept_defaults behaviour should not overwrite previous answers.

If the commands happens a first time, without accept defaults, and then a second time, accept-defaults usually takes the previous answers? I might be wrong

Reply:
The user might want to replace the previously defined mappings non-interactively and pass "--accept-defaults" to avoid interactive prompts. Then the mappings from the spec would be applied.

... or we can keep the previous mappings and merge them with the ones from the spec. Then the only way to remove pci specs would be to use the interactive prompts.

I guess there are pros and cons either way.

Reply:
I've updated the non-interactive code path to use the manifest as main source of truth, if a manifest was provided and stick to the previous answers otherwise.